.github/workflows/tests.yml

```diff
@@ -0,0 +1,47 @@
+name: Tests
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "main"
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    permissions:
+      # Gives the action the necessary permissions for publishing new
+      # comments in pull requests.
+      pull-requests: write
+      # Gives the action the necessary permissions for pushing data to the
+      # python-coverage-comment-action branch, and for editing existing
+      # comments (to avoid publishing multiple comments in the same PR)
+      contents: write
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
There are v4 of the checkout and upload-artifact actions that have library updates and speed improvements.  Probably not needed, but trying to keep those updated when I have to modify actions.

Reply:
Will update.